File: codegen/results/v002/qwen3.6_35b_nothink/20260418_170552/python/007_beatles_interview/solution.py
import csv
import json
from datetime import datetime

def calculate_age(birthday_str, reference_date):
    """Calculate age in years as of reference_date from birthday_str (MM/DD/YYYY)."""
    birth_date = datetime.strptime(birthday_str, "%m/%d/%Y")
    age = reference_date.year - birth_date.year
    # Check if the birthday hasn't occurred yet in the reference year
    if (reference_date.month, reference_date.day) < (birth_date.month, birth_date.day):
        age -= 1
    return age

def main():
    reference_date = datetime(2025, 7, 1)
    
    # Read the CSV file
    input_csv_path = 'input/input.csv'
    output_json = []
    
    with open(input_csv_path, mode='r', newline='') as csvfile:
        reader = csv.DictReader(csvfile)
        for row in reader:
            # Parse the Name field into First and Last Name
            name_parts = row['Name'].strip().split()
            if len(name_parts) >= 2:
                first_name = name_parts[0]
                last_name = ' '.join(name_parts[1:])
            elif len(name_parts) == 1:
                first_name = name_parts[0]
                last_name = ''
            else:
                first_name = ''
                last_name = ''
            
            birthday_str = row['Birthday'].strip()
            age = calculate_age(birthday_str, reference_date)
            
            relatives = []
            
            # Father
            father_name = row['Father'].strip()
            if father_name and father_name.lower() != 'null':
                parts = father_name.split()
                if parts:
                    relatives.append({
                        "FirstName": parts[0],
                        "LastName": ' '.join(parts[1:]) if len(parts) > 1 else '',
                        "Relationship": "Father"
                    })
            
            # Mother
            mother_name = row['Mother'].strip()
            if mother_name and mother_name.lower() != 'null':
                parts = mother_name.split()
                if parts:
                    relatives.append({
                        "FirstName": parts[0],
                        "LastName": ' '.join(parts[1:]) if len(parts) > 1 else '',
                        "Relationship": "Mother"
                    })
            
            # Brother
            brother_name = row['Brother'].strip()
            if brother_name and brother_name.lower() != 'null':
                parts = brother_name.split()
                if parts:
                    relatives.append({
                        "FirstName": parts[0],
                        "LastName": ' '.join(parts[1:]) if len(parts) > 1 else '',
                        "Relationship": "Brother"
                    })
            
            # Sister
            sister_name = row['Sister'].strip()
            if sister_name and sister_name.lower() != 'null':
                parts = sister_name.split()
                if parts:
                    relatives.append({
                        "FirstName": parts[0],
                        "LastName": ' '.join(parts[1:]) if len(parts) > 1 else '',
                        "Relationship": "Sister"
                    })
            
            person = {
                "FirstName": first_name,
                "LastName": last_name,
                "Birthday": birthday_str,
                "Age": age,
                "Relatives": relatives
            }
            
            output_json.append(person)
    
    # Output the JSON array to stdout
    print(json.dumps(output_json, indent=2))

if __name__ == "__main__":
    main()

File: codegen/results/v002/qwen3.6_35b_nothink/20260418_170552/typescript/007_beatles_interview/solution.ts
import * as fs from 'fs';
import * as path from 'path';

function calculateAge(birthdayStr: string, referenceDate: Date): number {
  const parts = birthdayStr.split('/');
  const month = parseInt(parts[0], 10);
  const day = parseInt(parts[1], 10);
  const year = parseInt(parts[2], 10);

  const birthday = new Date(year, month - 1, day);
  
  let age = referenceDate.getFullYear() - birthday.getFullYear();
  const monthDiff = referenceDate.getMonth() - birthday.getMonth();
  const dayDiff = referenceDate.getDate() - birthday.getDate();
  
  if (monthDiff < 0 || (monthDiff === 0 && dayDiff < 0)) {
    age -= 1;
  }
  
  return age;
}

function main() {
  const inputPath = path.join('input', 'input.csv');
  const content = fs.readFileSync(inputPath, 'utf-8');
  
  const lines = content.trim().split('\n');
  const headers = lines[0].split(',');
  
  const referenceDate = new Date(Date.UTC(2025, 6, 1)); // July 1, 2025
  
  const result: any[] = [];
  
  for (let i = 1; i < lines.length; i++) {
    const line = lines[i];
    // Split by comma, but need to handle potential issues. Assuming no commas in fields based on sample.
    const values = line.split(',');
    
    const nameStr = values[0].trim();
    const birthdayStr = values[1].trim();
    // const died = values[2].trim(); // Not used
    
    const fatherStr = values[3].trim();
    const motherStr = values[4].trim();
    const brotherStr = values[5].trim();
    const sisterStr = values[6].trim();
    
    // Parse name: "FirstName LastName"
    // Some names have middle names like "John Winston Lennon". 
    // The expected output shows FirstName: "John", LastName: "Lennon".
    // So first word is FirstName, last word is LastName.
    const nameParts = nameStr.split(' ');
    const firstName = nameParts[0];
    const lastName = nameParts[nameParts.length - 1];
    
    const age = calculateAge(birthdayStr, referenceDate);
    
    const relatives: any[] = [];
    
    if (fatherStr && fatherStr !== 'null') {
      const fParts = fatherStr.split(' ');
      relatives.push({
        FirstName: fParts[0],
        LastName: fParts[fParts.length - 1],
        Relationship: "Father"
      });
    }
    
    if (motherStr && motherStr !== 'null') {
      const mParts = motherStr.split(' ');
      relatives.push({
        FirstName: mParts[0],
        LastName: mParts[mParts.length - 1],
        Relationship: "Mother"
      });
    }
    
    if (brotherStr && brotherStr !== 'null') {
      const bParts = brotherStr.split(' ');
      relatives.push({
        FirstName: bParts[0],
        LastName: bParts[bParts.length - 1],
        Relationship: "Brother"
      });
    }
    
    if (sisterStr && sisterStr !== 'null') {
      const sParts = sisterStr.split(' ');
      relatives.push({
        FirstName: sParts[0],
        LastName: sParts[sParts.length - 1],
        Relationship: "Sister"
      });
    }
    
    result.push({
      FirstName: firstName,
      LastName: lastName,
      Birthday: birthdayStr,
      Age: age,
      Relatives: relatives
    });
  }
  
  console.log(JSON.stringify(result, null, 2));
}

main();